tests/imodels-access-backend-tests/src/BackendiModelsAccess.test.ts

```diff
@@ -181,6 +183,45 @@ describe("BackendIModelsAccess", () => {
     });
   });
 
+  describe("CreateNewIModel", () => {
+    it("copyAndPrepareBaselineFile should perform a wal checkpoint", async () => {
+      // cspell:disable-next-line
+      const filePath = path.join(testDownloadPath, "createnewimodel.bim");
+      const walPath = `${filePath}-wal`;
+      const loggerSpy = sinon.spy(Logger, "logWarning").withArgs("BackendIModelsAccess", "Wal file found while uploading file, performing checkpoint.", sinon.match.any);
+      await IModelHost.startup();
+      const testIModel = StandaloneDb.createEmpty(filePath, { rootSubject: { name: "test1"}, allowEdit: JSON.stringify({ txns: true })});
+      PhysicalModel.insert(testIModel, IModel.rootSubjectId, "TestModel");
+
+      expect(testIModel.isOpen).to.be.true;
+      expect(IModelJsFs.existsSync(walPath)).to.be.true;
+      expect(IModelJsFs.lstatSync(walPath)?.size).to.be.greaterThan(0);
+      const arg: CreateNewIModelProps = {
+        version0: filePath,
+        iModelName: "testimodel",
+        iTwinId: Guid.createValue()
+      };
+      // Expect database locked because we didn't save changes to end our transaction.
+      expect(() => (backendIModelsAccess as any).copyAndPrepareBaselineFile(arg)).to.throw("database is locked");
+      expect(IModelJsFs.lstatSync(walPath)?.size).to.be.greaterThan(0);
+      testIModel.saveChanges(); // end txn.
+      let tempBaselineFilePath = (backendIModelsAccess as any).copyAndPrepareBaselineFile(arg);
```

Comment:
this is a rather long test, seems a bit hard to read for me. it looks like the code is split into logical blocks by whitespace but I personally cannot figure out what's the think that each of the blocks do. could you add some comments to highlight the workflow that is being tested?

Reply:
I've attempted to make the test more readable, let me know.

Reply:
looks great, especially the link to docs on what is a wal file to begin with. thanks!